src/postgres/postgresConnectionStrings.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ConnectionOptions, parse } from "pg-connection-string";
+import { postgresDefaultPort } from "../constants";
+import { ParsedConnectionString } from "../ParsedConnectionString";
+import { nonNullProp } from "../utils/nonNull";
+
+export function parsePostgresConnectionString(connectionString: string): ParsedPostgresConnectionString {
+    const config: ConnectionOptions = parse(connectionString.trim());
+    return new ParsedPostgresConnectionString(connectionString, config);
+}
+
+export function createPostgresConnectionString(host: string, port: number = postgresDefaultPort, username?: string | undefined, password?: string | undefined): ParsedPostgresConnectionString {
+    let connectionString: string = `postgres://`;
+    if (username && password) {
+        connectionString += `${username}:${password}@`;
+    }
+    connectionString += `${host}:${port}`;
+    return parsePostgresConnectionString(connectionString);
+}
+
+export function fixedEncodeURIComponent(component: string): string {
+    return encodeURIComponent(component).replace(/[!'()*]/g, escape);
```

Comment:
Why are you doing this? If you're going to do something weird (like modifying the Node.js default "encodeURIComponent"), you need to explain why and probably link to docs

Reply:
You still haven't answered why you need to escape extra characters?

Reply:
I put a comment in the new commit. So, encodeURIComponent does not escape `A-Z a-z 0-9 - _ . ! ~ * ' ( )`, which include characters that are allowed for passwords.

Reply:
Yes I read your comment - why do you need to encode those characters?